### ISSUE TITLE
Async media: move pending posts / media upload handling from UploadService to FluxC

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#6352]
In the `UploadService`  in[ branch `feature/async-media`](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/async-media/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java)
These can benefit from being implemented in a table in FluxC:
```
- private static final ConcurrentHashMap<Integer, List<MediaModel>> sCompletedMediaByPost = new ConcurrentHashMap<>();
- private static final List<PostModel> sPostsWithPendingMedia = new ArrayList<>();
- private static final ConcurrentHashMap<Integer, UploadError> sFailedUploadPosts
- private static final ConcurrentHashMap<Integer, List<MediaModel>> sFailedMediaByPost = new ConcurrentHashMap<>();
```
This way we would always keep current information about progress and what has been uploaded  / is pending upload or is failed, and potentially set the basis for process recovery at a later point.

Same for media upload progress (although we'd need to check whether server-side upload resuming is in place for this to make sense)

**Comment by mzorz** [comment]
related #6282

**Comment by aforcier** [comment]
Addressed in #6613.
